Automattic/jetpack
# Ensure block comments are always sanitized on save and render

Fixes #
139-gh-Automattic/verbum
p3btAN-2AX-p2

## Proposed changes:
* This PR moves rendering and filtering logic out of `class-verbum-gutenberg-editor.php` to `class-verbum-comments`. The gutenberg editor is loaded conditionally at the moment. Because of this the filter `pre_comment_content` to which we attach the sanitization function is not executed properly.
* Our sanitization function was not considering `innerBlocks` this PR updates it to recursively sanitize the blocks.
 
### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
- Sync this PR to your sandbox and an atomic site.
- Load a post from your sites and load the comment form without the block editor.
- Post this as a comment ``.
- Remove the disabled attribute from the submit button and submit the form.
- Load the same post but this time with the block editor enabled.
- The latest-posts block should be removed from the comment.
